app/services/badges/award_contributor_from_github.rb

```diff
@@ -25,6 +25,8 @@ def initialize(msg)
     end
 
     def call
+      return unless Settings::Authentication.providers.include?(:github)
```

Comment:
Very nice! 💯  I just checked and it caches nicely, so it won't even put any load on the DB.